src/libraries/System.Net.Security/src/System/Net/Security/SslStream.IO.cs

```diff
@@ -236,6 +240,11 @@ private async Task RenegotiateAsync<TIOAdapter>(CancellationToken cancellationTo
                     _buffer.ReturnBuffer();
                 }
 
+                if (token.Payload != null)
```

Comment:
Move the null check for payload inside of clear and always call it.

Reply:
the check was already inside. I just removed the extra check and in some places I moved it out of branches to make it more clear that that cleanup would happen.